tests/build/build.go

```diff
@@ -516,6 +517,250 @@ var _ = framework.BuildSuiteDescribe("Build service E2E tests", Label("build", "
+
+			// Delete created webhook from GitHub
+			hooks, err := f.AsKubeAdmin.CommonController.Github.ListRepoWebhooks(helloWorldComponentGitSourceRepoName)
+			Expect(err).NotTo(HaveOccurred())
+
+			for _, h := range hooks {
+				hookUrl := h.Config["url"].(string)
+				if strings.Contains(hookUrl, pacControllerHost) {
+					Expect(f.AsKubeAdmin.CommonController.Github.DeleteWebhook(helloWorldComponentGitSourceRepoName, h.GetID())).To(Succeed())
+					break
+				}
+			}
+		})
+
+		When("component is created", func() {
+			It("triggers a pipeline run", func() {
+				Expect(f.AsKubeAdmin.HasController.WaitForComponentPipelineToBeFinished(component, "", 2)).To(Succeed())
+				Expect(f.AsKubeAdmin.TektonController.DeleteAllPipelineRunsInASpecificNamespace(testNamespace)).To(Succeed())
+			})
+
+			Specify("simple build can be triggered manually", func() {
+				Expect(f.AsKubeAdmin.HasController.SetComponentAnnotation(componentName, controllers.BuildRequestAnnotationName, controllers.BuildRequestTriggerSimpleBuildAnnotationValue, testNamespace)).To(Succeed())
+
+				Expect(f.AsKubeAdmin.HasController.WaitForComponentPipelineToBeFinished(component, "", 2))
+				Expect(f.AsKubeAdmin.TektonController.DeleteAllPipelineRunsInASpecificNamespace(testNamespace))
+			})
+		})
+
+		Context("using PaC", func() {
+			BeforeAll(func() {
+				if utils.IsPrivateHostname(consoleRoute.Spec.Host) {
+					Skip("Using private cluster (not reachable from Github), skipping...")
+				}
+			})
+
+			It("configures PaC via annotation", func() {
+				Expect(f.AsKubeAdmin.HasController.SetComponentAnnotation(componentName, controllers.BuildRequestAnnotationName, controllers.BuildRequestConfigurePaCAnnotationValue, testNamespace)).To(Succeed())
+
+				var buildStatus *controllers.BuildStatus
+
+				Eventually(func() (bool, error) {
+					component, err := f.AsKubeAdmin.HasController.GetComponent(componentName, testNamespace)
+					if err != nil {
+						GinkgoWriter.Printf("cannot get the component: %v\n", err)
+						return false, err
+					}
+
+					statusBytes := []byte(component.Annotations[controllers.BuildStatusAnnotationName])
+
+					err = json.Unmarshal(statusBytes, &buildStatus)
+					if err != nil {
+						GinkgoWriter.Printf("cannot unmarshal build status: %v\n", err)
+						return false, err
+					}
+
+					if buildStatus.PaC != nil {
+						GinkgoWriter.Printf("state: %s\n", buildStatus.PaC.State)
+						GinkgoWriter.Printf("mergeUrl: %s\n", buildStatus.PaC.MergeUrl)
+						GinkgoWriter.Printf("errId: %d\n", buildStatus.PaC.ErrId)
+						GinkgoWriter.Printf("errMessage: %d\n", buildStatus.PaC.ErrMessage)
+						GinkgoWriter.Printf("configurationTime: %d\n", buildStatus.PaC.ConfigurationTime)
+					} else {
+						GinkgoWriter.Println("build status does not have PaC field")
+					}
+
+					return buildStatus.PaC != nil && buildStatus.PaC.State == "enabled" && buildStatus.PaC.MergeUrl != "" && buildStatus.PaC.ErrId == 0 && buildStatus.PaC.ConfigurationTime != "", nil
```

Comment:
Will it be good to similar keep a check for simple build as well like `buildStatus.Simple != nil`

Reply:
@tnevrlka We are combining simple build and PaC build together.. I think `buildStatus.Simple != nil` check should be checked when we are checking if the simple build was successful.